src/main/java/com/znsio/teswiz/runner/BrowserDriverManager.java

```diff
@@ -195,25 +206,171 @@ private static DriverManagerType setupBrowserDriver(TestExecutionContext testExe
                                                         String browserType) {
         DriverManagerType driverManagerType = DriverManagerType.valueOf(browserType.toUpperCase());
         String webDriverManagerProxyUrl = (null == Runner.getWebDriverManagerProxyURL()) ? ""
-                                                                                         :
-                                          Runner.getWebDriverManagerProxyURL();
+                :
+                Runner.getWebDriverManagerProxyURL();
         LOGGER.info(String.format(
                 "Using webDriverManagerProxyUrl: '%s' for getting the WebDriver for browser: '%s'",
                 webDriverManagerProxyUrl, browserType));
 
         // TODO - get browser version from local or container. What about cloud?
+
+        String localBrowserVersion = getLocalBrowserVersionFor(browserType);
+        String containerBrowserVersion = getContainerBrowserVersion();
+
         WebDriverManager webDriverManager = WebDriverManager.getInstance(driverManagerType)
-                                                            .proxy(webDriverManagerProxyUrl);
+                .proxy(webDriverManagerProxyUrl);
         webDriverManager.setup();
         String downloadedDriverVersion = webDriverManager.getDownloadedDriverVersion();
 
         String message = String.format("Using %s browser version: %s", driverManagerType,
-                                       downloadedDriverVersion);
+                downloadedDriverVersion);
         LOGGER.info(message);
         ReportPortalLogger.logInfoMessage(message);
         return driverManagerType;
     }
 
+    private static String getContainerBrowserVersion() {
+        DefaultDockerClientConfig config = DefaultDockerClientConfig.createDefaultConfigBuilder().build();
+        DockerClient dockerClient = DockerClientBuilder.getInstance(config).build();
+
+        String imageName = "";
+        String containerId = "";
+
+        try {
+            Yaml yaml = new Yaml();
+            FileInputStream fis = new FileInputStream("docker-compose.yml");
+            Map<String, Object> data = yaml.load(fis);
+
+            Map<String, Object> services = (Map<String, Object>) data.get("services");
+            Map<String, Object> selenium = (Map<String, Object>) services.get("chrome");
+            imageName = (String) selenium.get("image");
+            LOGGER.info("Selenium image: " + imageName);
+        } catch (FileNotFoundException e) {
+            e.printStackTrace();
+        }
+
+        List<Container> containers = dockerClient.listContainersCmd()
+                .withShowAll(true)
+                .exec();
+        for (Container container : containers) {
+            if (container.getImage().equals(imageName)) {
+                containerId = container.getId();
+            }
+        }
+
+        assert containerId != null;
```

Comment:
What is the use of this assert here
and why are we passing it here

Reply:
if the container object does not find the matching id for container name it will return null. the assertion is added for the same.